### DISCUSSION TITLE
Copying non ts files to build folder

### DISCUSSION THREAD

**Discussion by iammrsea** [adonisjs/core#1397] [General]
Typically, typescript ignores non ts files such as .json, .yml during build. This turned out to be a problem for me while I was working a project with V5. I had a documentation folder which contained a .yml file, an OpenApi3 documentation for my API, and I needed this folder to be copied over to the build directory during build time so that it will be deployed with the rest of my application. I solved this problem by writing an ace command that copied the documentation folder to the build directory after build, 'postbuild'. The ace command was just a wrapper around 'copyfiles' package. It worked fine.

However, while working on my current project, it turned out there's a better way to achieve the same thing without writing any command or installing any package for it. All I did was just to register the path to the folder I wanted to be copied over to build directory in .adonisrc.json file and adonis takes over from there. Snippet:

```javascript
//.adonisrc.json
"metaFiles":[
...,
"documentation/**"
]
```

**Comment by thetutlage** [comment]
And if you don't want the server to restart when changing certain files, then you can define it as follows:

```json
{
  "metaFiles": [
    {
      "pattern": "documentation/**",
	  "reloadServer": false

    }
  ]
}
```

**Reply by iammrsea** [reply]
Thank you for that tip.
In addition, is there any way I can add a glob pattern in exceptRoutes in config/shield.ts?
I tried something like this:
```javascript
//config/shield.ts
exceptRoutes: ["/api/v1/**"]
```
but it didn't work. It only worked when I write the route in full. I have quite a few routes to except because my app is both web and api server for a mobile app, and it's a pain to register all the routes individually.

**Reply by thetutlage** [reply]
Two different topics. I suggest opening a new discussion thread for it

**Reply by iammrsea** [reply]
Okay
